lib/client/player.go

```diff
@@ -50,7 +57,8 @@ type sessionPlayer struct {
 	sessionEvents []events.EventFields
 	term          *terminal.Terminal
 
-	// stopC is used to tell the caller that player has finished playing
+	// stopC is called when playback ends (either because the end of the stream has
+	// been reached, or a hard stop was requested via EndPlayback().
 	stopC    chan int
```

Comment:
It this is just for closing, then consider this instead:

```suggestion
	stopC    chan struct{}
```

Another Q: do we need both a state and closeC channel? Could we get away with a state for "closed" and just the lock/sync.Cond?

Reply:
struct{} causes no allocations. It also makes clear that the value being sent doesn't matter, whereas an int could mean a number of things.

> I think so. There are other changes we should make to this player as well, however it is low priority and so we are just trying to pragmatically fix the functionality with these changes.